### PR TITLE
Put non-ignorable extenders ahead of ignorable extenders

### LINKED ISSUE #78258: Put non-ignorable extenders ahead of ignorable extenders
**What would you like to be added**:
Put non-ignorable extenders ahead of ignorable extenders in genericScheduler#extenders

**Why is this needed**:
In genericScheduler#processPreemptionWithExtenders, we iterate over genericScheduler#extenders.
If an ignorable extender gives error, the error is ignored and loop continues.

It seems we should place non-ignorable extenders ahead of ignorable extenders so that the preemption can bail out early if a non-ignorable extender gives error.

### PR DESCRIPTION
**Pull request by tedyu**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
In genericScheduler#processPreemptionWithExtenders, we iterate over genericScheduler#extenders.
If an ignorable extender gives error, the error is ignored and loop continues.

It seems we should place non-ignorable extenders ahead of ignorable extenders so that the preemption can bail out early if a non-ignorable extender gives error.

**Which issue(s) this PR fixes**:
Fixes #78258

```release-note
NONE
```

### CODE REVIEW

File: pkg/scheduler/factory/factory.go [hunk 1]
```diff
@@ -400,6 +400,19 @@ func (c *configFactory) CreateFromConfig(policy schedulerapi.Policy) (*Config, e
...
 		}
+		// place non-ignorable extenders ahead of ignorable extenders
+		var sortedExtenders []algorithm.SchedulerExtender
+		for i := range extenders {
+			if !extenders[i].IsIgnorable() {
+				sortedExtenders = append(sortedExtenders, extenders[i])
+			}
+		}
+		for i := range extenders {
+			if extenders[i].IsIgnorable() {
```
Comment (JieJhih) [inline pkg/scheduler/factory/factory.go]: Why not put this to above loop?

**Review by LexCC — CHANGES_REQUESTED** [review]

File: pkg/scheduler/factory/factory.go (hunk 1 above)
Comment (tedyu) [reply pkg/scheduler/factory/factory.go]: Folded into main loop

File: pkg/scheduler/factory/factory.go [hunk 2]
```diff
@@ -387,19 +387,27 @@ func (c *configFactory) CreateFromConfig(policy schedulerapi.Policy) (*Config, e
...
 			klog.V(2).Infof("Creating extender with config %+v", policy.ExtenderConfigs[ii])
 			extender, err := core.NewHTTPExtender(&policy.ExtenderConfigs[ii])
 			if err != nil {
 				return nil, err
 			}
-			extenders = append(extenders, extender)
+			if extender.IsIgnorable() {
+				extenders = append(extenders, extender)
+			} else {
+				sortedExtenders = append(sortedExtenders, extender)
```
Comment (Huang-Wei) [inline pkg/scheduler/factory/factory.go]: It's actually not literally "sorted" - it's two types of extenders. I'd suggest to use `ignorableExtenders` to store ignorable entenders, and append nonIgnorableExtender to `extenders`.

And also L409~L410 should be changed to:

```go
extenders = append(extenders, ignorableExtenders...)
```

Comment (tedyu) [reply pkg/scheduler/factory/factory.go]: Yeah - that would be cleaner.

Addressed

File: pkg/scheduler/factory/factory.go [hunk 3]
```diff
@@ -387,19 +387,26 @@ func (c *configFactory) CreateFromConfig(policy schedulerapi.Policy) (*Config, e
...
+			} else {
+				ignorableExtenders = append(ignorableExtenders, extender)
+			}
 			for _, r := range policy.ExtenderConfigs[ii].ManagedResources {
 				if r.IgnoredByScheduler {
 					ignoredExtendedResources.Insert(string(r.Name))
 				}
 			}
 		}
+		// place non-ignorable extenders ahead of ignorable extenders
```
Comment (Huang-Wei) [inline pkg/scheduler/factory/factory.go]:
```suggestion
		// place ignorable extenders to the tail of extenders
```

**Review by Huang-Wei — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**